util/gen_action_info.py

```diff
@@ -0,0 +1,250 @@
+    next(csv_file)
+    # Second line is the headers
+    # Append the hexidecimal version of the ID as a new column so the original ID is usable for correllation
+    headers = next(csv_file) + ["HexID"]
+    next(csv_file)
+
+    # Change # to ID for more readable data
+    headers[0] = headers[0].replace("#", "ID")
+    # Generate the hex ID from the ID and return the data with the headers prepended
+    return [headers] + [x + [format(int(x[0]), "X")] for x in csv_file]
+
+
+def __get_local_table(filename, inputs, outputs=None):
+    """Gets table data from a local file"""
+    # I'm honestly not even sure if this will work as expected. It can be updated/fixed if a use case is found.
+    with open(filename, "r") as table:
+        csv_file = [x.rstrip("\r\n").split(",") for x in table.read().decode("utf-8-sig")]
+
+    # First line is the indices and third line is the data types.  They aren't currently used, so we discard them.
+    csv_file.pop(2)
+    csv_file.pop(0)
+
+    # Next line is the headers
+    # Append the hexidecimal version of the ID as a new column so the original ID is usable for correllation
+    headers = csv_file.pop(0) + ["HexID"]
+    # Change # to ID for more readable data
+    headers[0] = headers[0].replace("#", "ID")
+    # Generate the hex ID from the ID and return the data with the headers prepended
+    return [headers] + [x + [format(int(x[0]), "X")] for x in csv_file]
+
+
+def get_data_table(table_name, locale="intl", inputs=None, outputs=None):
+    """Retrieves table data based on provided locale"""
+    if locale == "local" and os.path.exists(table_name):
+        return __get_local_table(table_name + ".csv", inputs, outputs)
+    if locale in config["locale_url"]:
+        url = config["locale_url"]["root"] + config["locale_url"][locale] + table_name + ".csv"
+        return __get_remote_table(url, inputs, outputs)
+    else:
+        raise Exception("Invalid locale: %s" % locale)
+
+
+def normalize_name(str):
+    """Converts names into JavaScript-safe ascii string keys that adhere to the expected conventions."""
+    if not str:
+        return str
+
+    # The Tam\u2013Tara Deepcroft
+    str = re.sub(r"\u2013", "-", str)
+
+    # The <Emphasis>Whorleater</Emphasis> Extreme
+    str = re.sub(r"</?Emphasis>", "", str)
+
+    # Various symbols to get rid of.
+    str = re.sub(r"[':(),]", "", str)
+
+    # Sigmascape V4.0 (Savage)
+    str = re.sub(r"\.", "", str)
+
+    # Common case hyphen: TheSecondCoilOfBahamutTurn1
+    # Special case hyphen: ThePalaceOfTheDeadFloors1_10
+    str = re.sub(r"([0-9])-([0-9])", r"\1_\2", str)
+    str = re.sub(r"[-]", " ", str)
+
+    # Of course capitalization isn't consistent, that'd be ridiculous.
+    str = str.title()
+
+    # collapse remaining whitespace
+    str = re.sub(r"\s+", "", str)
+
+    # remove non-ascii characters
+    str = re.sub(r"[^0-9A-z_]", "", str)
+
+    return str
+
+
+def write_js(filename, scriptname, variable, d):
```

Comment:
This looks like a duplicate of CoinachWriter.write.  It's not a good name at this point, but I don't think it should be duplicated.

Reply:
Same thing as the csv stuff, I think it can be improved.  Even moreso than simple structure, though, I'm not even going to begin to think that I should change what is there (yet).

You guys are obviously doing something right, since cactbot is already awesome, so I'm not arrogant enough to think that I can take one look at it and say it's wrong.

Reply:
If it can be improved, then I think the correct approach is to improve it, and not to duplicate it because you don't want to touch the original.  It's not a question of arrogance, but more just a willingness to take responsibility and make changes that other people can use.

(Also, not "guys")